### ISSUE TITLE
panic: runtime error when trying to establish a new channel

### ISSUE THREAD

**Issue by cpsola** [lightningnetwork/lnd#112]
When trying to create a new channel with the [testnet LN faucet](http://faucet.lightning.community/), I get this error:

```
alice     | panic: runtime error: invalid memory address or nil pointer dereference
alice     | [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4cf4ff]
alice     | 
alice     | goroutine 27 [running]:
alice     | panic(0xb1b3c0, 0xc420016080)
alice     | 	/usr/local/go/src/runtime/panic.go:500 +0x1a1
alice     | github.com/lightningnetwork/lnd/lnwallet.(*LightningChannel).StateSnapshot(0x0, 0x0)
alice     | 	/go/src/github.com/lightningnetwork/lnd/lnwallet/channel.go:2131 +0x2f
alice     | main.newChanAnnouncement(0xc420176660, 0x0, 0xb00109075, 0xc420bb0000, 0xc4204207c0, 0xc4204207c0, 0x2)
alice     | 	/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:610 +0x40
alice     | main.(*fundingManager).announceChannel(0xc42005ce00, 0xc4205bdb80, 0x0, 0xb00109075, 0x0, 0xc4204207c0, 0xc4204207c0)
alice     | 	/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:803 +0x78
alice     | main.(*fundingManager).handleFundingOpen(0xc42005ce00, 0xc420420930)
alice     | 	/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:857 +0x472
alice     | main.(*fundingManager).reservationCoordinator(0xc42005ce00)
alice     | 	/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:264 +0x2ed
alice     | created by main.(*fundingManager).Start
alice     | 	/go/src/github.com/lightningnetwork/lnd/fundingmanager.go:185 +0xce
alice exited with code 2

```
The error seems to appear when the opening transaction is confirmed.

I'm using docker containers created from the Dockerfiles in this repo.

**Comment by AndrewSamokhvalov** [comment]
@cpsola what version of `lnd` are you using?

**Comment by cpsola** [comment]
`[INF] LTND: Version 0.1.1-alpha`

Do you need any other info?

Thnks

**Comment by AndrewSamokhvalov** [comment]
@cpsola yeah, I think it would be great to get all `lnd` logs, it might facilitate investigation.

**Comment by cpsola** [comment]
lncli listpeers
{
	"peers": [
		{
			"pub_key": "036a0c5ea35df8a528b98edf6f290b28676d51d0fe202b073fe677612a39c0aa09",
			"peer_id": 1,
			"address": "159.203.125.125:10011",
			"bytes_sent": 5270,
			"bytes_recv": 24158
		}
	]
}

**Comment by cpsola** [comment]
@AndrewSamokhvalov Ok, I reproduced all the procedure again.

I had a docker container with btcd and another container with lnd. 

I connected the lnd with the testnet faucet peer:

`lncli listpeers`
`{
	"peers": [
		{
			"pub_key": "036a0c5ea35df8a528b98edf6f290b28676d51d0fe202b073fe677612a39c0aa09",
			"peer_id": 1,
			"address": "159.203.125.125:10011",
			"bytes_sent": 5270,
			"bytes_recv": 24158
		}
	]
}`

And introduced the nodes identity_pubkey key in the [faucet website](http://faucet.lightning.community/):
`030da942ed7cfc7d3096811b3264e15115778e692eaacb2b7a76fb27a58cbb5359`

After the [opening transaction](https://testnet.smartbit.com.au/tx/34107170a08a69a65bd0c218155db3d004bbd4f4bd65118dfdd537caffbc062c) confirmed, the logs on the lnd docker container showed the same error. I'm attaching the full lnd container log. At the begining it shows an error when looking for lnd.conf (I don't know if this is related).
[docker-alice-lnd-log.txt](https://github.com/lightningnetwork/lnd/files/722048/docker-alice-lnd-log.txt)

Let me know if you need anything else!

**Comment by AndrewSamokhvalov** [comment]
@cpsola Thanks for reproducing it again! The first thing which strikes the eye is that `btcd` isn't fully synced:
```
CRTR: Block 0000000013daee31b3b789c45e5978ce77fea61e2b4fadfc398af5121eb1bc75 (height=1038135) closed 0 channels
```

**Comment by cpsola** [comment]
@AndrewSamokhvalov Ok, I'll wait until btcd is fully syncronized and then try to create the channel again.

**Comment by Roasbeef** [comment]
The panic you ran into has been partially addressed due to de70175be60b22a45506dfea600d9f42a19b69bb. The currently missing portion would be to _disallow_ all channel creation until `lnd` has fully synced to the chain.

**Comment by cpsola** [comment]
Indeed, waiting for btcd to fully syncronize solved the problem.

I am now able to create the channel without problems :) :

`# lncli listchannels {
	"channels": [
		{
			"remote_pubkey": "036a0c5ea35df8a528b98edf6f290b28676d51d0fe202b073fe677612a39c0aa09",
			"channel_point": "1cfd66eba99cc17febd90e1ca5c12c6888915e15e11edd461d08f9800a7fb923:0",
			"chan_id": 1193774958648688640,
			"capacity": 3005000,
			"local_balance": 2000000,
			"remote_balance": 1000000
		}
	]
}`

Thnks for the quick responses!

**Comment by AndrewSamokhvalov** [comment]
@cpsola Glad it helped! As @Roasbeef said the issues had been partially addressed, and additional work still need to be done, so leave it open.

**Comment by Roasbeef** [comment]
The second portion of the issue has been fixed by a658fabf481a0ede6e94d860bf19fff49bc0d66c, so I'll be closing this issue now.
